3p/ampcontext.js

```diff
@@ -138,4 +141,14 @@ export class AmpContext {
     }
     return ancestors[(ancestors.length - 1) - depth];
   }
+
+  /**
+   *  Checks to see if there is a window variable assigned with the
+   *  sentinel value, sets it, and returns true if so.
+   *  @private
+   */
+  getWindowSentinel_() {
+    this.sentinel = this.win_.AMP_SENTINEL || null;
```

Comment:
this is to allow a bypass with safeframe if needed, in which case we don't want to write on the sentinel or hash. instead we can write a script into the ad itself that sets this variable